cms/release_calendar/forms.py

```diff
@@ -0,0 +1,91 @@
+from datetime import datetime
+from typing import Any
+
+from django.core.exceptions import ValidationError
+from django.utils.translation import gettext as _
+from wagtail.admin.forms import WagtailAdminPageForm
+from wagtail.models import Locale
+
+from .enums import NON_PROVISIONAL_STATUS_CHOICES, ReleaseStatus
+
+
+class ReleaseCalendarPageAdminForm(WagtailAdminPageForm):
+    """Convenience administrative form for easier calendar page data validation."""
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+
+        if self.instance.status != ReleaseStatus.PROVISIONAL:
+            # Once a release calendar page is confirmed, it cannot go back to provisional
+            self.fields["status"].choices = NON_PROVISIONAL_STATUS_CHOICES
+
+        if self.instance.status == ReleaseStatus.PUBLISHED:
+            self.fields["release_date"].disabled = True
+
+    def clean(self) -> dict:
+        """Validate the submitted release calendar data."""
+        cleaned_data: dict = super().clean()
+
+        status = cleaned_data["status"]
+
+        if status == ReleaseStatus.CANCELLED and not cleaned_data["notice"]:
+            raise ValidationError({"notice": _("The notice field is required when the release is cancelled")})
```

Comment:
General point: some error messages are being marked for localisation, and some aren't. I suggest we mark all and be consistent.